### PR TITLE
[git-webkit] Make the default rebase behavior for `git-webkit pr` configurable

### PR DESCRIPTION
**Pull request by JonWBedard**
#### b4c270a0742517728c686c774f30d33f0ac4c547
<pre>
[git-webkit] Make the default rebase behavior for `git-webkit pr` configurable
<a href="https://bugs.webkit.org/show_bug.cgi?id=239385">https://bugs.webkit.org/show_bug.cgi?id=239385</a>
&lt;rdar://problem/91811534 &gt;

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.main): Check the webkitscmpy.auto-rebase-branch option to determine
if we should automatically rebase the branch.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.git): Prompt user to set the automatic rebase behavior.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:

Canonical link: <a href="https://commits.webkit.org/251475@main">https://commits.webkit.org/251475@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295470">https://svn.webkit.org/repository/webkit/trunk@295470</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by graouts** [comment]
Removed myself as a reviewer, I don't have the required expertise to do so.

**Comment by JonWBedard** [comment]
> Removed myself as a reviewer, I don't have the required expertise to do so.

I think the better question for you @graouts is does this change improve your workflow, since you're the one that requested it.

**Comment by graouts** [comment]
If it lets me set a default that calling `git webkit pr` does _not_ yield a rebase, then the answer is a resounding **YES**.

**Comment by mrego** [comment]
Rebasing by default is very time consuming and it's the opposite to the default workflow with `webkit-patch`, shouldn't we change the default too? Apart from making it configurable?

**Comment by marcoscaceres** [comment]
I found this (automatic rebase) surprising also... any chance we can revive this discussion?... and, ironically, rebase this patch after fixing the style issue 😜

**Comment by JonWBedard** [comment]
> I found this (automatic rebase) surprising also... any chance we can revive this discussion?... and, ironically, rebase this patch after fixing the style issue 😜

The style issue is from the unit tests, going to make the code less clear if we try and fix it. Fell off my queue in the last few weeks, but re-pinging reviewers, change still applies and definitely seems relevant for folks making changes to WebCore and code lower in the stack.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py [hunk 1]
```diff
@@ -259,7 +259,14 @@ def create_pull_request(cls, repository, args, branch_point):
             sys.stderr.write("'{}' is not a remote in this repository\n".format(source_remote))
             return 1
 
-        rebasing = args.rebase or (args.rebase is None and repository.config().get('pull.rebase'))
+        if args.rebase is not None:
+            rebasing = args.rebase or (args.rebase is None and repository.config().get('pull.rebase'))
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py]: `(args.rebase is None and repository.config().get('pull.rebase'))` will always be false as this is in `args.rebase is not None:` condition.
So we can just do 
```
 rebasing = args.rebase is not None or repository.config().get(
                 'webkitscmpy.auto-rebase-branch',
                 repository.config().get('pull.rebase', 'true'),
             ) == 'true
```

**Review by dewei-zhu — APPROVED** [review]
r=me with comment.

**Pull request merged**